src/test/java/org/openrewrite/java/testing/assertj/MigrateChainedAssertToAssertJTest.java

```diff
@@ -0,0 +1,439 @@
+
+        String finalArgument = firstArg.equals("") ? secondArg : firstArg;
+        finalArgument = finalArgument.contains(".") ? finalArgument.split("\\.")[0] : finalArgument;
+
+        String before = String.format(template, formattedAssertBefore);
+        List<String> formattedArgs = new ArrayList<>(Arrays.asList(dedicatedAssertion, finalArgument));
+        if (!firstArg.equals("") && !secondArg.equals("")) {
+            formattedArgs.add(secondArg);
+        }
+        String after = String.format(template, assertAfter.formatted(formattedArgs.toArray()));
+
+        rewriteRun(
+          java(before, after)
+        );
+    }
+
+    private static Stream<Arguments> optionalReplacements() {
+        return Stream.of(
+          Arguments.arguments("isPresent", "isTrue", "isPresent", ""),
+          Arguments.arguments("get", "isEqualTo", "contains", "something"),
+          Arguments.arguments("get", "isSameAs", "containsSame", "something")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("optionalReplacements")
+    void optionalReplacements(String chainedAssertion, String assertToReplace, String dedicatedAssertion, String arg ) {
+        //language=java
+        String template = """
+        import org.junit.jupiter.api.Test;
+        import java.util.Optional;
+        
+        import static org.assertj.core.api.Assertions.assertThat;
+        
+        class MyTest {
+            @Test
+            void test() {
+                String something = "hello world";
+                %s
+            }
+            
+            Optional<String> getOptional() {
+                return Optional.of("hello world");
+            }
+        }
+        """;
+
+        String assertBefore = String.format("assertThat(getOptional().%s()).%s(%s);", chainedAssertion, assertToReplace, arg);
+        String assertAfter = String.format("assertThat(getOptional()).%s(%s);", dedicatedAssertion, arg);
+
+        String before = String.format(template, assertBefore);
+        String after = String.format(template, assertAfter);
+
+        rewriteRun(java(before, after));
+    }
+
+    private static Stream<Arguments> arrayReplacements() {
+        return Stream.of(
+          Arguments.arguments("isZero", "isEmpty", ""),
+          Arguments.arguments("isEqualTo", "hasSize", "length"),
+          Arguments.arguments("isEqualTo", "hasSameSizeAs", "anotherArray.length"),
+          Arguments.arguments("isLessThanOrEqualTo", "hasSizeLessThanOrEqualTo", "expression"),
+          Arguments.arguments("isLessThan", "hasSizeLessThan", "expression"),
+          Arguments.arguments("isGreaterThan", "hasSizeGreaterThan", "expression"),
+          Arguments.arguments("isGreaterThanOrEqualTo", "hasSizeGreaterThanOrEqualTo", "expression")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("arrayReplacements")
+    @Disabled("array.length is not a method invocation, so that's not matched or handled just yet")
```

Comment:
As indicated `array.length` is not a method invocation, so it fails quite early in the checks. Not sure if we'd want to cover this within the same recipe already; perhaps best to leave this out and potentially later pick that up separately.

Reply:
Probably best to remove these tests and the commented out recipes as well then to really clean up the PR before a merge.